test/forcing/ForcingsEngineGriddedDataProvider_Test.cpp

```diff
@@ -0,0 +1,124 @@
+#include <gtest/gtest.h>
+
+#ifndef NGEN_FORCINGS_GRIDDED_CONFIG_PATH
+#error "Gridded config file path not defined! Set `-DNGEN_FORCINGS_GRIDDED_CONFIG_PATH`. " \
+       "Note: you should not be seeing this compile error. Please contact an ngen developer."
+#endif
+
+#include "ForcingsEngineTestHelpers.hpp"
+
+#include <forcing/ForcingsEngineGriddedDataProvider.hpp>
+
+struct ForcingsEngineGriddedDataProviderTest
+  : public ForcingsEngineDataProviderTest
+{
+    using provider_type = data_access::ForcingsEngineGriddedDataProvider;
+
+    ForcingsEngineGriddedDataProviderTest()
+      : ForcingsEngineDataProviderTest()
+    {}
+
+    static std::unique_ptr<provider_type> provider_;
+
+    static void SetUpTestSuite();
+};
+
+using TestFixture = ForcingsEngineGriddedDataProviderTest;
+
+std::unique_ptr<TestFixture::provider_type> TestFixture::provider_ = nullptr;
+
+constexpr const char* config_file = NGEN_FORCINGS_GRIDDED_CONFIG_PATH;
+
+constexpr auto cat_11223_mask = geojson::box_t{
+  /*min_corner=*/{ -71.0554067, 43.1423006 },
+  /*max_corner=*/{ -70.9668753, 43.185111 }
+};
+
+constexpr auto cat_11410_mask = geojson::box_t{
+  /*min_corner=*/{ -71.0577593, 43.1375584 },
+  /*max_corner=*/{ -71.0210404, 43.1761626 }
+};
+
+void TestFixture::SetUpTestSuite()
+{
+    std::cout << "Initializing (Gridded) ForcingsEngineDataProviderTest" << std::endl;
+    ForcingsEngineDataProviderTest::SetUpTestSuite();
+
+    TestFixture::time_start = data_access::detail::parse_time("2023-01-17 01:00:00");
+    TestFixture::time_end   = TestFixture::time_start + 3600 + 3600;
+
+    TestFixture::provider_ = std::make_unique<data_access::ForcingsEngineGriddedDataProvider>(
+        /*init=*/config_file,
+        /*time_begin_seconds=*/TestFixture::time_start,
+        /*time_end_seconds=*/TestFixture::time_end,
+        /*mask=*/cat_11223_mask
+    );
+
+    #if NGEN_WITH_MPI
+    auto comm = MPI_Comm_c2f(MPI_COMM_WORLD);
```

Comment:
I think this should be explicitly typed for clarity that we care about the specific type
```suggestion
    MPI_Fint comm = MPI_Comm_c2f(MPI_COMM_WORLD);
```